src/Processors/QueryPlan/Optimizations/optimizeDirectReadFromTextIndex.cpp

```diff
@@ -36,6 +36,39 @@ String getNameWithoutAliases(const ActionsDAG::Node * node)
     return node->result_name;
 }
 
+String optimizationInfoToString(const IndexReadColumns &added_columns, const Names &removed_columns)
+{
+    chassert(!added_columns.empty());
+
+    String result = "Added: [";
+
+    // This will list the index and the new associated columns
+    size_t idx = 0;
+    for (const auto &[_, name_and_type] : added_columns)
```

Comment:
This must be plural and less ambiguous. `name_and_type` --> `columns_names_and_types`